templates/dex-k8s-authenticator.yaml

```diff
@@ -34,22 +34,26 @@ spec:
         hosts:
           - ""
       dexK8sAuthenticator:
-        port: 5555
-        debug: false
-        web_path_prefix: /token
         #logoUrl: http://<path-to-your-logo.png>
         #tlsCert: /path/to/dex-client.crt
         #tlsKey: /path/to/dex-client.key
         clusters:
         - name: kubernetes-cluster
           short_description: "Kubernetes cluster"
           description: "Kubernetes cluster authenticator"
-          # client_secret: value is generated automatically - see `setupDexK8sAuthenticator()`
-          # This `client_id` must by in sync with `dex.yaml` value as well as
-          # kube-apiserver flag `oidc-client-id`.
+          # client_secret: value is generated automatically via initContainers
           client_id: kube-apiserver
           issuer: https://dex-kubeaddons.kubeaddons.svc.cluster.local:8080/dex
-          # This URI is just a placeholder and it will be replaced during the
-          # deployment with URL pointing to the traefik ingress public
-          # load balancer.
+          # This URI is just a placeholder and it will be replaced during initContainers
+          # with a URL pointing to the traefik ingress public load balancer.
           redirect_uri: https://dex-k8s-authenticator-kubeaddons.kubeaddons.svc.cluster.local:5555/token/callback/kubernetes-cluster
+      caCerts:
+        enabled: true
+        secrets:
+        - name: dex-root-ca
+          filename: dex-root-ca.crt
+          value: cGxhY2Vob2xkZXIK # b64 "placeholder" this will be updated by initcontainers
```

Comment:
`value: cGxhY2Vob2xkZXIK ` I think this is wrong...

Reply:
```
$ echo 'cGxhY2Vob2xkZXIK' |base64 -d
placeholder
```

Reply:
Why do we need in first place to add this secret ?

Reply:
That should be done by the init container, right ?

Reply:
that is what I assume instead of having to add a secret with values that will be overwritten afterwards.

Reply:
This is related to https://github.com/kubernetes/kubernetes/issues/35183, upstream Kubernetes does not currently have a solid mechanism for the opposite of an initContainer, and so the full lifecycle of the secret resource can not be managed by the underlying deployment. Adding it here makes helm responsible for the end-of-life of this resource and so it will be cleaned up properly as per helm delete.

Reply:
This is ugly 🤦‍♂. I am not envisioning a scenario where someone installs dex and then uninstalls it to keep a cluster working 🤔 .  I am still thinking this should be done by the init-container, but I won't block you.

Reply:
In this particular case, this is explicitly done via values to workaround a deficiency in the helm chart. [DCOS-56979](https://jira.mesosphere.com/browse/DCOS-56979) is a follow up to deal with this deficiency as part of later iterations. This should help with the ugliness @hectorj2f.

Reply:
Talked to hector in Slack, going to consider this resolved.